stdlib/Mmap/src/Mmap.jl

```diff
@@ -212,6 +212,7 @@ function mmap(io::IO,
     @static if Sys.isunix()
         prot, flags, iswrite = settings(file_desc, shared)
         iswrite && grow && grow!(io, offset, len)
+        !iswrite && mmaplen > filesize(io) && throw(ArgumentError("unable to mmap file size smaller than $mmaplen due to read-only permissions. Please check your dims"))
```

Comment:
```suggestion
        szfile = convert(Csize_t, len + offset)
        !iswrite && szfile > filesize(io) && throw(ArgumentError("unable to increase file size to $szfile due to read-only permissions"))
```
(stealing from windows, though we can potentially DRY some of the code here)